optuna/visualization/_contour.py

```diff
@@ -143,7 +154,11 @@ def _get_contour_plot(study: Study, params: Optional[List[str]] = None) -> "go.F
 
 
 def _generate_contour_subplot(
-    trials: List[FrozenTrial], x_param: str, y_param: str, direction: StudyDirection
+    trials: List[FrozenTrial],
+    x_param: str,
+    y_param: str,
+    direction: StudyDirection,
+    param_values_range: dict,
```

Comment:
```suggestion
    param_values_range: Dict[str, Tuple[float, float]],
```

We have to additionally import `from typing import Dict`.

Reply:
And, how about making this argument's default value `None`? It is more easy to use this function when testing. If this is `None`, then the `param_values_range` is set to be `{}` (empty dictionary).
```suggestion
    param_values_range: Optional[Dict[str, Tuple[float, float]]] = None,
```
In line 163, we need
```python
if param_values_range is None:
    param_values_range = {}
```